src/pages/api/stream/[name].js

```
@@ -0,0 +1,209 @@
+      code: 403,
+      message: 'You are not authorized',
+      data: undefined
+    });
+
+  const name = request?.params?.name;
+
+  if (request.method.toUpperCase() !== 'POST') {
+    reply.code(400).send({
+      success: false,
+      code: 400,
+      message: 'Method is not allowed'
+    });
+  }
+
+  /**
+   * @typedef Config
+   * @property {string} api_key - A string of key that will be used to access inLive protected API
+   */
+
+  /**
+   * A variable to create object
+   *
+   * @param {Config} config -- being passed from init module parameter
+   */
+  let config = {
+    api_key: process.env.API_KEY
+  };
+
+  /**
+   * Initialize an initialization instance
+   *
+   * @function
+   * @param {Config} config - A set of key/value parameter configuration
+   * @returns {object} InitializationInstance that contains config object of api_key
+   */
+  const inliveApp = InliveApp.init(config);
+
+  switch (name) {
+    case 'create': {
+      try {
+        let dataStream;
+
+        const configObject = {
+          name: request?.body?.name || '',
+          description: request?.body?.description
+            ? request?.body?.description || ''
+            : ''
+        };
+
+        if (configObject.name !== undefined && configObject.name !== null) {
+          // trigger create function
+          dataStream = await InliveStream.createStream(inliveApp, configObject);
+        }
+
+        if (dataStream.status.code === 200) {
+          reply.code(200).send({
+            success: true,
+            code: dataStream.status.code,
+            message: dataStream.status.message,
+            data: dataStream.data
+          });
+        } else {
+          reply.code(dataStream.status.code).send({
+            success: false,
+            code: dataStream.status.code,
+            message: dataStream.status.message,
+            data: dataStream.data
+          });
+        }
+      } catch (error) {
+        throw new Error(error);
```

`throw new Error(error);` will expect a string. Just `throw error` without `new Error()` will throw the actual error data response from API request which can be an object to the client/front-end. The front-end will handle that error instead. @ingrid011 @diahasumsi

---

@faiq-naufal @diahasumsi I think with `throw error` can be applicable since the error data response from API request will be sent as an object in the future. For now, if the code change like that, seems still fine (same workable).

---

I just realized we don't need to `catch` the error if we just want to throw the error to the client. Fastify already handles that. Fastify will log the error on the server side and throw the error to the client side.